### PR TITLE
Supplement as_*_array functions

### PR DESCRIPTION
**Pull request by findepi**
These functions are useful when implementing UDFs. It would be helpful for downstream projects to provide graceful cast functions for all data types.

### CODE REVIEW

File: datafusion/common/src/cast.rs [hunk 1]
```diff
@@ -93,14 +131,9 @@ pub fn as_string_view_array(array: &dyn Array) -> Result<&StringViewArray> {
     Ok(downcast_value!(array, StringViewArray))
 }
 
-// Downcast ArrayRef to UInt32Array
-pub fn as_uint32_array(array: &dyn Array) -> Result<&UInt32Array> {
```
Comment (alamb) [inline datafusion/common/src/cast.rs:97]: these were just moved elsewhere in this file I think

**Review by alamb — APPROVED** [review]
Looks like a nice improvement to me. Thank you @findepi

File: datafusion/common/src/cast.rs (hunk 1 above)
Comment (findepi) [reply datafusion/common/src/cast.rs:97]: yes, i moved 2-3 functions to make this file a bit more ordered. maybe i should have abstained from that

**Review by Weijun-H — APPROVED** [review]
LGTM! Thanks @findepi. This pr makes the code more organized.

**Comment by Weijun-H** [comment]
Thanks @alamb for reviewing

**Pull request merged**